src/mcp/java/net/minecraftforge/gradle/mcp/function/PatchFunction.java

```diff
@@ -20,105 +20,73 @@
 
 package net.minecraftforge.gradle.mcp.function;
 
-import com.cloudbees.diff.PatchException;
-import net.minecraftforge.gradle.common.diff.ContextualPatch;
-import net.minecraftforge.gradle.common.diff.ContextualPatch.PatchReport;
-import net.minecraftforge.gradle.common.diff.HunkReport;
-import net.minecraftforge.gradle.common.diff.PatchFile;
-import net.minecraftforge.gradle.common.diff.ZipContext;
+import codechicken.diffpatch.gradle.patch.PatchAction;
+import codechicken.diffpatch.util.PatchMode;
 import net.minecraftforge.gradle.common.util.HashStore;
+import net.minecraftforge.gradle.common.util.Utils;
 import net.minecraftforge.gradle.mcp.util.MCPEnvironment;
-import org.apache.commons.io.IOUtils;
 
 import java.io.File;
 import java.io.IOException;
-import java.util.List;
+import java.nio.file.Files;
 import java.util.Map;
-import java.util.Map.Entry;
-import java.util.stream.Collectors;
 import java.util.zip.ZipFile;
 
 public class PatchFunction implements MCPFunction {
 
     private String path;
-    private Map<String, String> patches;
+    private File patchesDir;
 
     @Override
-    public  void loadData(Map<String, String> data) {
+    public void loadData(Map<String, String> data) {
         path = data.get("patches");
     }
 
     @Override
-    public  void initialize(MCPEnvironment environment, ZipFile zip) throws IOException {
-        patches = zip.stream().filter(e -> !e.isDirectory() && e.getName().startsWith(path) && e.getName().endsWith(".patch"))
-        .collect(Collectors.toMap(e -> e.getName().substring(path.length()), e -> {
-            try {
-                return IOUtils.toString(zip.getInputStream(e));
-            } catch (IOException ex) {
-                throw new RuntimeException(ex);
-            }
-        }));
+    public void initialize(MCPEnvironment environment, ZipFile zip) throws IOException {
+        patchesDir = new File(environment.getWorkingDir(), "patches");
+        Utils.extractDirectory(e -> new File(patchesDir, e.substring(path.length())), zip, path);
     }
```

Comment:
I specifically did not want to extract these files, as it's a lot of small disc churn. Our entire process is about working in zip files, not exploded directories.